examples/k8s-health-check/proxy_with_http_health_check.yaml

```diff
@@ -49,90 +49,104 @@ spec:
-          # - "--private-ip"
+        # Replace <INSTANCE_CONNECTION_NAME> with the instance connection
+        # name in the format: "project_name:region:instance_name"
+        # Replace <DB_PORT> with the port that the proxy should open
+        # to listen for database connections from the application
+        - <INSTANCE_CONNECTION_NAME>
 
-          # Enable HTTP health checks
-          - "--health-check"
+        env:
+        # Using environment variables instead of CLI arguments to configure the
+        # proxy k8s configuration can make it easier to read your
+        # k8s configuration files.
+        #
+        # This is the recommended configuration for your proxy pod template.
+        # It is used by the cloud-sql-proxy-operator when configuring the
+        # proxy container.
 
-          # Listen on all addresses so the kubelet can reach the endpoints
-          - "--http-address=0.0.0.0"
+        # Replace <DB_PORT> with the port that the proxy should open
+        # to listen for database connections from the application
+        - name: CSQL_PROXY_PORT
+          value: <DB_PORT>
 
-          # Set the port where the HTTP server listens
-          # - "--http-port=9090"
+        # If connecting from a VPC-native GKE cluster, you can use the
+        # following flag to have the proxy connect over private IP
+        # - name: CSQL_PROXY_PRIVATE_IP
+        #   value: "true"
 
-          # Enable structured logging with LogEntry format:
-          - "--structured-logs"
+        # Enable HTTP healthchecks on port 9801. This enables /liveness,
+        # /readiness and /startup health check endpoints. Allow connections
+        # listen for connections on any interface (0.0.0.0) so that the
+        # k8s management components can reach these endpoints.
+        - name: CSQL_PROXY_HEALTH_CHECK
+          value: "true"
+        - name: CSQL_PROXY_HTTP_PORT
+          value: "9801"
+        - name: CSQL_PROXY_HTTP_ADDRESS
+          value: 0.0.0.0
 
-          # This flag specifies where the service account key can be found
-          # Remove this argument if you are using workload identity
-          - "--credentials-file=/secrets/service_account.json"
+        # Configure the proxy to exit gracefully when sent a k8s configuration
+        # file.
+        - name: CSQL_PROXY_EXIT_ZERO_ON_SIGTERM
+          value: "true"
 
-          # Replace DB_PORT with the port the proxy should listen on
-          - "--port=<DB_PORT>"
-          - "<INSTANCE_CONNECTION_NAME>"
+        # Enable the admin api server (which only listens for local connections)
+        # and enable the /quitquitquit endpoint. This allows other pods
+        # to shut down the proxy gracefully when they are ready to exit.
+        - name: CSQL_PROXY_QUITQUITQUIT
+          value: "true"
+        - name: CSQL_PROXY_ADMIN_PORT
+          value: "9092"
+        - name: CSQL_PROXY_USER_AGENT
+          value: cloud-sql-proxy-operator/unknown
```

Comment:
I don't think we should add this here -- this is a private implementation detail for our purposes here.

Reply:
Good catch, fixed.